ArcGISRuntimeSDKQt_CppSamples/EditData/SnapGeometryEdits/SnapGeometryEdits.h

```diff
@@ -0,0 +1,118 @@
+
+  Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(bool geometryEditorStarted READ geometryEditorStarted NOTIFY geometryEditorStartedChanged)
+  Q_PROPERTY(bool canUndo READ canUndo NOTIFY canUndoChanged)
+  Q_PROPERTY(bool elementIsSelected READ elementIsSelected NOTIFY elementIsSelectedChanged)
+  Q_PROPERTY(bool layersLoaded MEMBER m_layersLoaded NOTIFY layersLoadedChanged)
+  Q_PROPERTY(QList<QString> pointLayers MEMBER m_pointLayers NOTIFY pointLayersChanged)
+  Q_PROPERTY(QList<QString> polylineLayers MEMBER m_polylineLayers NOTIFY polylineLayersChanged)
+  Q_PROPERTY(QList<bool> pointSourceCheckedState MEMBER m_pointSourceCheckedState NOTIFY pointSourceCheckedStateChanged)
+  Q_PROPERTY(QList<bool> polylineSourceCheckedState MEMBER m_polylineSourceCheckedState NOTIFY polylineSourceCheckedStateChanged)
+
+public:
+  explicit SnapGeometryEdits(QObject* parent = nullptr);
+  ~SnapGeometryEdits() override;
+
+  enum class GeometryEditorMode {
+    PointMode,
+    MultipointMode,
+    PolylineMode,
+    PolygonMode
+  };
+
+  Q_ENUM(GeometryEditorMode)
+
+  static void init();
+  Q_INVOKABLE void startEditor(GeometryEditorMode geometryEditorMode);
+  Q_INVOKABLE void stopEditor();
+  Q_INVOKABLE void deleteSelection();
+  Q_INVOKABLE void editorUndo();
+  Q_INVOKABLE void snappingEnabledStatus(bool checkedValue);
+  Q_INVOKABLE void onPointLayersEnabled();
+  Q_INVOKABLE void onPolylineLayersEnabled();
+  Q_INVOKABLE void displaySnapSources();
+  Q_INVOKABLE void pointSourceEnabledStatus(bool snappingCheckedState, int index);
+  Q_INVOKABLE void polylineSourceEnabledStatus(bool snappingCheckedState, int index);
+
+signals:
+  void canUndoChanged();
+  void elementIsSelectedChanged();
+  void geometryEditorStartedChanged();
+  void layersLoadedChanged();
+  void mapViewChanged();
+  void pointLayersChanged();
+  void pointSourceCheckedStateChanged();
+  void polylineLayersChanged();
+  void polylineSourceCheckedStateChanged();
+
+private:
+  Esri::ArcGISRuntime::MapQuickView* mapView() const;
+  void setMapView(Esri::ArcGISRuntime::MapQuickView* mapView);
+  bool geometryEditorStarted() const;
+  bool canUndo();
+  void createInitialSymbols();
+  void createConnections();
+  bool elementIsSelected();
+
+
+
+  Esri::ArcGISRuntime::Map* m_map = nullptr;
+  Esri::ArcGISRuntime::MapQuickView* m_mapView = nullptr;
+  Esri::ArcGISRuntime::GraphicsOverlay* m_graphicsOverlay = nullptr;
+  Esri::ArcGISRuntime::Graphic* m_editingGraphic = nullptr;
+  Esri::ArcGISRuntime::GeometryEditor* m_geometryEditor = nullptr;
+  Esri::ArcGISRuntime::SimpleMarkerSymbol* m_pointSymbol = nullptr;
+  Esri::ArcGISRuntime::SimpleMarkerSymbol* m_multiPointSymbol = nullptr;
+  Esri::ArcGISRuntime::SimpleLineSymbol* m_lineSymbol = nullptr;
+  Esri::ArcGISRuntime::SimpleFillSymbol* m_polygonSymbol = nullptr;
+  QList<bool> m_pointSourceCheckedState;
+  QList<bool> m_polylineSourceCheckedState;
```

Comment:
Are these the best data structures for this? If there are external UI components they likely have their own "checked" state you can query. If these are simply duplicating that state, it doesn't sound like a good idea.